superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -34,27 +45,170 @@ export const FormFieldOrder = [
   'username',
   'password',
   'database_name',
+  'credentials_info',
+  'query',
+  'encryption',
 ];
 
 interface FieldPropTypes {
   required: boolean;
+  hasTooltip?: boolean;
+  tooltipText?: (valuse: any) => string;
+  onParametersChange: (value: any) => string;
+  onParametersUploadFileChange: (value: any) => string;
   changeMethods: { onParametersChange: (value: any) => string } & {
     onChange: (value: any) => string;
-  };
+  } & { onParametersUploadFileChange: (value: any) => string };
   validationErrors: JsonObject | null;
   getValidation: () => void;
+  db?: DatabaseObject;
+  isEditMode?: boolean;
+  sslForced?: boolean;
+  defaultDBName?: string;
+  editNewDb?: boolean;
 }
 
+const CredentialsInfo = ({
+  changeMethods,
+  isEditMode,
+  db,
+  editNewDb,
+}: FieldPropTypes) => {
+  const [uploadOption, setUploadOption] = useState<number>(
+    CredentialInfoOptions.jsonUpload.valueOf(),
+  );
+  const [fileToUpload, setFileToUpload] = useState<string | null | undefined>(
+    null,
+  );
+  return (
+    <CredentialInfoForm>
+      {!isEditMode && (
+        <>
+          <FormLabel required>
+            How do you want to enter service account credentials?
```

Comment:
We need to make these strings (here and below) translatable (with `t()`).

Reply:
I _think_ I got them all in [`this commit`](https://github.com/apache/superset/pull/15463/commits/b8be120b60f52982f7135c222828b0426c322faf).